test/permission_test.js

```diff
@@ -121,45 +189,77 @@ describe("canCreateRecord", () => {
       const collection = {permissions: {"record:create": ["chucknorris"]}};
 
-      expect(canCreateRecord(session, collection)).eql(false);
+      expect(canCreateRecord(session, bucket, collection)).eql(false);
     });
   });
 
-  describe("write permission", () => {
+  describe("collection write permission", () => {
     it("should check if a record can be created in a collection", () => {
+      const bucket = {permissions: {write: ["chucknorris"]}};
       const collection = {permissions: {write: [1]}};
 
-      expect(canCreateRecord(session, collection)).eql(true);
+      expect(canCreateRecord(session, bucket, collection)).eql(true);
     });
+  });
 
-    it("should check if a record cannot be created in a collection", () => {
+  describe("bucket write permission", () => {
+    it("should check if a record can be created in a collection", () => {
+      const bucket = {permissions: {write: [1]}};
       const collection = {permissions: {write: ["chucknorris"]}};
 
-      expect(canCreateRecord(session, collection)).eql(false);
+      expect(canCreateRecord(session, bucket, collection)).eql(true);
     });
   });
 });
 
 describe("canEditRecord", () => {
   const session = {authenticated: true, serverInfo: {user: {id: 1}}};
 
-  it("should check if a record can be edited", () => {
-    const record = {permissions: {write: [1]}};
+  describe("record write permission", () => {
+    it("should check if a record can be edited", () => {
+      const bucket = {permissions: {write: ["chucknorris"]}};
+      const collection = {permissions: {write: ["chucknorris"]}};
+      const record = {permissions: {write: [1]}};
+
+      expect(canEditRecord(session, bucket, collection, record)).eql(true);
+    });
+
+    it("should check if a record cannot be edited", () => {
+      const bucket = {permissions: {write: ["chucknorris"]}};
+      const collection = {permissions: {write: ["chucknorris"]}};
+      const record = {permissions: {write: ["chucknorris"]}};
+
+      expect(canEditRecord(session, bucket, collection, record)).eql(false);
+    });
+  });
+
+  describe("collection write permission", () => {
+    it("should check if a record can be edited", () => {
+      const bucket = {permissions: {write: ["chucknorris"]}};
+      const collection = {permissions: {write: [1]}};
+      const record = {permissions: {write: ["chucknorris"]}};
 
-    expect(canEditRecord(session, record)).eql(true);
+      expect(canEditRecord(session, bucket, collection, record)).eql(true);
+    });
   });
 
-  it("should check if a record cannot be edited", () => {
-    const record = {permissions: {write: ["chucknorris"]}};
+  describe("bucket write permission", () => {
+    it("should check if a record can be edited", () => {
+      const bucket = {permissions: {write: [1]}};
+      const collection = {permissions: {write: ["chucknorris"]}};
+      const record = {permissions: {write: ["chucknorris"]}};
```

Comment:
General remark in this test suite: for better understandability, it may be worth switching from `chucknorris` and `1` to `current-user` and `someone-else`. WDYT?